### PR TITLE
Match PtrToStringUTF8 to the other PtrToString* families.

### LINKED ISSUE #23723: Marshal.PtrToStringUTF8(IntPtr, int) has a different behavior
The `Marshal.PtrToStringUTF8(IntPtr, int)` method has a different behavior as compared to the other `Marshal.PtrToString*(IntPtr, int)` methods.

In particular, the other `PtrToString` methods will throw if `ptr == IntPtr.Zero`. However, `PtrToStringUTF8` will return `null`.

We should resolve these to be consistent.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fixes #23723.

I decided to keep `PtrToStringUTF8` throwing an `ArgumentOutOfRangeException` so that we were consistent there with prior behavior, and since an `ArgumentOutOfRangeException` inherits from `ArgumentException` it still fulfills the contract for the wrapping `PtrToStringAuto`.

### CODE REVIEW

**Comment by tannergooding** [comment]
> since an ArgumentOutOfRangeException inherits from ArgumentException it still fulfills the contract for the wrapping PtrToStringAuto.

Maybe we should fix all of them; for consistency?

**Comment by AaronRobinsonMSFT** [comment]
cc @GrabYourPitchforks

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs [hunk 1]
```diff
@@ -57,7 +57,7 @@ public static unsafe string PtrToStringAnsi(IntPtr ptr, int len)
             }
             if (len < 0)
             {
-                throw new ArgumentException(null, nameof(len));
+                throw new ArgumentOutOfRangeException(nameof(len), SR.ArgumentOutOfRange_NeedNonNegNum);
```
Comment (hughbe) [inline src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: nit: you can pass `len` to the `(string, object, string)` constructor

**Comment by jkotas** [comment]
Could you please submit PR into CoreFX with the required test updates that can be cherry picked once this change reaches CoreFX?

**Comment by jkotas** [comment]
https://github.com/dotnet/corefx/pull/36612

**Pull request merged**